### ISSUE TITLE
Metaclass conflict while importing pydantic under `freezegun.freeze_time`

### ISSUE THREAD

**Issue by vbarbaresi** [pydantic/pydantic#4983]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://pydantic-docs.helpmanual.io) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Importing pydantic while under `freezegun.freeze_time()` is crashing with:

```python
Traceback (most recent call last):
  File "/tmp/pydantic_bug_repro.py", line 7, in <module>
    import pydantic
  File "pydantic/__init__.py", line 2, in init pydantic.__init__
  File "pydantic/dataclasses.py", line 43, in init pydantic.dataclasses
    # +---------+-----------------------------------------+
  File "pydantic/error_wrappers.py", line 4, in init pydantic.error_wrappers
  File "pydantic/json.py", line 14, in init pydantic.json
  File "pydantic/types.py", line 1162, in init pydantic.types
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

This is relatively minor and to be fair I don't know if the issue is more on freezegun side or on pydantic's side.
This didn't happen on pydantic==1.9.x and started happening on 1.10.x

Note: freezegun package is https://pypi.org/project/freezegun/

### Example Code

```Python
import datetime
from freezegun import freeze_time


freeze_dt = datetime.datetime.now()
with freeze_time(freeze_dt):
    import pydantic
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.4
            pydantic compiled: True
                 install path: /Users/vincent.barbaresi/go/src/github.com/DataDog/dd-analytics/dd-ve3/lib/python3.8/site-packages/pydantic
               python version: 3.8.16 (default, Dec  7 2022, 01:24:57)  [Clang 14.0.0 (clang-1400.0.29.202)]
                     platform: macOS-12.6.2-arm64-arm-64bit
     optional deps. installed: ['typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://pydantic-docs.helpmanual.io/changelog/)
- [ ] [Data validation/parsing](https://pydantic-docs.helpmanual.io/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://pydantic-docs.helpmanual.io/usage/exporting_models/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://pydantic-docs.helpmanual.io/usage/schema/)
- [ ] [Dataclasses](https://pydantic-docs.helpmanual.io/usage/dataclasses/)
- [ ] [Model Config](https://pydantic-docs.helpmanual.io/usage/model_config/)
- [ ] [Field Types](https://pydantic-docs.helpmanual.io/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/)
- [ ] [Generic Models](https://pydantic-docs.helpmanual.io/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://pydantic-docs.helpmanual.io/usage/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [X] [Plugins](https://pydantic-docs.helpmanual.io/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by hramezani** [comment]
Thanks @vbarbaresi for reporting.

As the error says it's because of the conflict in metaclasses.

When you are using the `freeze_time`, the `datetime.date` will be replaced with [freezegun.api.FakeDate](https://github.com/spulec/freezegun/blob/9d5ea73e0464e5856bc7ab75cc45b35d7eb88c32/freezegun/api.py#L307) class that has [FakeDateMeta](https://github.com/spulec/freezegun/blob/9d5ea73e0464e5856bc7ab75cc45b35d7eb88c32/freezegun/api.py#L280) as metaclass. This conflict with `ConstrainedDate` that has `ConstrainedNumberMeta` as metaclass.
https://github.com/pydantic/pydantic/blob/0bc7cd39a90fa46c23ca9ef9d19c51ecb8e1145c/pydantic/types.py#L1169

We are working on Pydantic `V2` and I am not sure what is happen for `ConstrainedDate` in `V2`. But we can accept a fix on `1.10.x` if it is an easy one.

**Comment by vbarbaresi** [comment]
It ended up quite minor and not blocking as we found a workaround:
importing pydantic before invoking `freezegun.freeze_time()` didn't trigger the conflict. (not sure why we don't hit the conflict in this case)
We were importing pydantic in the local scope inside `freeze_time()` for no good reason: this bug is an edge case

**Comment by washeck** [comment]
We are affected by this as well. I wouldn't say it is an minor edge case as it blocks us from using `frezegun` and `django-qr-code` together. The way it manifests is quite surprising, because depending on order of execution you get random test failures - in our case all tests are passing in CI but if you run one particular test (using `freezegun`), the test fails.

Thank you for the workaround (we added import of `pydantic` to our test settings to ensure it is always loaded) but a proper fix would be much welcome.

**Comment by samuelcolvin** [comment]
I doubt this will be easy to fix in V1.10, since it's caused by the new `ConstraintedDate` types added in #3740. I guess we could check if `freezegun` has already money patched `date`s metaclass before using our own, but that sounds ugly.

Best work around as you mentioned is to change the import order.

I've checked and this is not a problem in V2.
